atecc608a_se.c

```diff
@@ -208,20 +245,64 @@ static psa_status_t atecc608a_asymmetric_sign(psa_key_slot_number_t key_slot,
     return status;
 }
 
+psa_status_t atecc608a_asymmetric_verify(psa_key_slot_number_t key_slot,
+                                         psa_algorithm_t alg,
+                                         const uint8_t *p_hash,
+                                         size_t hash_length,
+                                         const uint8_t *p_signature,
+                                         size_t signature_length)
+{
+    const uint16_t key_id = key_slot;
+    psa_status_t status = PSA_ERROR_GENERIC_ERROR;
+    bool is_verified = false;
+
+    /* Keys 8 to 15 can store public keys. */
+    if (key_id < 7 || key_id > 15)
+    {
+        return PSA_ERROR_INVALID_ARGUMENT;
+    }
+
+    /* We can only do ECDSA on SHA-256 */
+    if (alg != PSA_ALG_ECDSA(PSA_ALG_SHA_256) && alg != PSA_ALG_ECDSA_ANY)
+    {
+        return PSA_ERROR_NOT_SUPPORTED;
+    }
+
+    if (hash_length != 32)
+    {
+        /* The driver only supports hashes of length 32. */
+        return PSA_ERROR_NOT_SUPPORTED;
+    }
+
+    if (signature_length != ATCA_SIG_SIZE)
+    {
+        /* The driver only supports signatures of length 64. */
+        return PSA_ERROR_NOT_SUPPORTED;
+    }
+
+    ASSERT_SUCCESS_PSA(atecc608a_init());
+
+    ASSERT_SUCCESS(atcab_verify_stored(p_hash, p_signature, key_id, &is_verified));
+
+exit:
+    atecc608a_deinit();
+    return status;
+}
 
 #define PSA_ATECC608A_LIFETIME 0xdeadbeefU
 
 static psa_drv_se_asymmetric_t atecc608a_asymmetric =
 {
     .p_sign = &atecc608a_asymmetric_sign,
-    .p_verify = 0,
+    .p_verify = &atecc608a_asymmetric_verify,
```

Comment:
The `&` here isn't necessary. I'd recommend removing it. I would also recommend removing it for `atecc608a_asymmetric_sign` in a separate commit.